src/System.Private.ServiceModel/tests/Scenarios/SelfHostWcfService/IWcfDuplexService.cs

```diff
@@ -31,4 +32,17 @@ public interface IDuplexChannelCallback
         [OperationContract(IsOneWay = true)]
         void OnPingCallback(Guid guid);
     }
+
+    [ServiceContract(CallbackContract = typeof(IWcfDuplexTaskReturnCallback))]
+    public interface IWcfDuplexTaskReturnService
+    {
+        [OperationContract]
+        Task<Guid> Ping(Guid guid);
+    }
+
+    public interface IWcfDuplexTaskReturnCallback
+    {
+        [OperationContract]
+        Task<Guid> ServicePingCallback(Guid guid);
```

Comment:
Yeah, I guess this is testing the request-reply path at the same time as the callback path, not a bad thing; let's just leave it as-is.